webapp/store/logic.py

```diff
@@ -33,21 +77,226 @@ def convert_date(date_to_convert):
+        "summary": rock["metadata"].get("summary", ""),
+        "icon_url": "",
+        "metadata": {
+            "license": rock["metadata"].get("license", ""),
+            "status": "chisel" if "chisel" in rock["name"] else "full",
+            "links": rock["metadata"].get("links", {}),
+            "private": rock["metadata"].get("private", False),
+            "rock_details": {
+                "repo_url": "",
+                "docs_url": "",
+                "SBOM_url": "",
+            },
+            "upstream_details": {},
+            "related_rocks": [],
+            "downstream_artifacts": {},
+        },
+        "categories": rock["metadata"].get("categories", []),
+        "publisher": {
+            "name": format_slug(
+                rock["metadata"]["publisher"].get("display-name", "")
+            ),
+            "username": rock["metadata"]["publisher"].get("username", ""),
+            "validation": rock["metadata"]["publisher"].get("validation", ""),
+        },
+        "channels": [],
+    }
+
+    parsed_rock["display_name"] = format_slug(rock.get("name", ""))
+    parsed_rock["icon_url"] = get_icon(rock["metadata"].get("media", []))
+    parsed_rock["license"] = rock["metadata"].get("license", "")
+
+    # Build channel info
+    for channel in rock.get("channel-map", []):
+        channel_data = channel.get("channel", {})
+        revision_data = channel.get("revision", {})
+        v = revision_data.get("version", "").split(".")
+        if len(v) < 3:
+            v.append("0")
+        normalized_version = ".".join(v)
+        parsed_channel = {
+            "workload_version": normalized_version,
+            "risk": channel_data.get("risk", ""),
+            "last_updated": format_relative_date(
+                channel_data.get("released-at", "")
+            ),
+            "released_at": convert_date(channel_data.get("released-at", "")),
+            "revision": revision_data["revision"],
+            "version": revision_data.get("version", ""),
+            "track": channel_data.get("track", ""),
+        }
+        parsed_rock["channels"].append(parsed_channel)
+        parsed_rock["latest_channel"] = max(
+            parsed_rock["channels"],
+            key=lambda x: datetime.datetime.strptime(
+                x["released_at"], "%d %b %Y"
+            ),
+        )
+    return parsed_rock
+
+
+def get_rocks(
+    size: int = 10,
+    query_params: Dict[str, Any] = {},
+) -> List[Dict[str, Any]]:
+    """
+    Fetches paginated and parsed rock packages using DeviceGW.
+    """
+    rocks2 = device_gw.find("%", fields=FIND_FIELDS).get("results", [])
+
+    total_items = len(rocks2)
+    total_pages = (total_items + size - 1) // size
+    page = int(query_params.get("page", 1))
+    rocks_per_page = paginate(rocks2, page, size)
+    parsed_rocks = []
+
+    for rock in rocks_per_page:
+        parsed_rocks.append(parse_package_for_card(rock))
```

Comment:
```suggestion
    parsed_rocks = [parse_package_for_card(rock) for rock in rocks_per_page]
```
This one is just a nitpick, but list comprehension is clearer IMO